src/Platform/OpenGL/Shaders/lens_flare.glsl

```diff
@@ -0,0 +1,34 @@
+layout(binding = 0) uniform sampler2D lensFlareColor;
+layout(binding = 1) uniform sampler2D inputColor;
+
+uniform int ghosts; // number of ghost samples
+uniform float ghostDispersal; // dispersion factor
+uniform float uHaloWidth;
+
+in vec2 TexCoord;
+out vec4 outColor;
+
+void main()
+{
+	vec2 texcoord = TexCoord;
+	texcoord = -texcoord + vec2(1.0);
+	// ghost vector to image centre:
+	vec2 ghostVec = (vec2(0.5) - texcoord) * ghostDispersal;
+
+	// sample ghosts
+	vec4 result = vec4(0.0);
+	for (int i = 0; i < ghosts; ++i) {
+		vec2 offset = fract(texcoord + ghostVec * float(i));
```

Comment:
is it some kind of noise? maybe use existing noise function?

Reply:
Nope I use fract() to generate repeating pattern for ghosts. It's similar to GL_REPEAT.